tests/test_admin.py

```diff
@@ -647,3 +649,163 @@ def test_cms_page_toolbar_pages_link_doesnt_filter_the_page_list_by_page_id(self
 
         self.assertSetEqual(set(response.context["cl"].queryset), set(pages))
         self.assertSetEqual(set(response_with_page_id.context["cl"].queryset), set(pages))
+
+
+class AdminSearchTestCase(CMSTestCase):
+    """
+    Test case covers custom search functionality.
+    """
+    def setUp(self):
+        template_1 = get_cms_setting('TEMPLATES')[0][0]
+        self.language = "en"
+        self.pagecontent = PageContentWithVersionFactory(
+            template=template_1, language=self.language,
+            title="This is a test"
+        )
+        PageContentWithVersionFactory.create_batch(
+            template=template_1, language=self.language,
+            size=5,
+        )
+        self.page_admin = PageContentAdmin(PageContent, admin)
+        # Use _base_manager so that non-published contents are available
+        self.page_queryset = PageContent._base_manager.all()
+        self.page_urls = []
+
+    def _get_page_admin_request(self, search_term):
+        return f"/admin/cms/pagecontent/?q={search_term}"
+
+    def test_page_url_search_partial_match_from_slug_and_path(self):
+        """
+        Partial URL matches to the slug and path return the PageContent associated with it.
+        """
+        PageUrlFactory(
+            page=self.pagecontent.page,
+            language=self.language,
+            path=slugify(self.pagecontent.title),
+            slug=slugify(self.pagecontent.title),
```

Comment:
There is a problem with these tests, because you are using the page title as the slug/path. As such you are not really testing the page is is found from the matching slug/path, because it would be found anyway because the title of the page includes `test` (you can see this by commenting out your new `get_search_results` method and running tests, these will still pass).

So it would be better to set a different slug/path, unrelated to the page title, which is closer to the original bug that was reported:
```suggestion
            slug=slugify("example-slug"),
```
And then update your request to search for this slug, rather than `"test"` which is in the title of the page so will be found anyway. With your `get_search_results` method commented out you will be able to see this fail first, then adding back in it should pass as expected.

Reply:
Good catch mate, will update now.